synapse/state/__init__.py

```diff
@@ -391,17 +391,14 @@ async def resolve_state_groups_for_events(
         state_group_ids_set = set(state_group_ids)
         if len(state_group_ids_set) == 1:
             (state_group_id,) = state_group_ids_set
-            state = await self._state_storage_controller.get_state_for_groups(
-                state_group_ids_set
-            )
             (
                 prev_group,
                 delta_ids,
             ) = await self._state_storage_controller.get_state_group_delta(
                 state_group_id
             )
             return _StateCacheEntry(
-                state=state[state_group_id],
+                state=None,
```

Comment:
I guess the point is that we don't need to fetch the entirety of the state here and now; we can just pass back a state group Id and let the caller pick out the specific pieces of state they need?